nni/tools/nnictl/config_utils.py

```diff
@@ -4,64 +4,106 @@
+        inverse_config['trial'] = {}
+        for kv in metadata_config:
+            if kv['key'] == 'local_config':
+                inverse_config['localConfig'] = kv['value']
+            elif kv['key'] == 'trial_config':
+                inverse_config['trial'] = kv['value']
+    elif platform == 'remote':
+        for kv in metadata_config:
+            if kv['key'] == 'machine_list':
+                inverse_config['machineList'] = kv['value']
+            elif kv['key'] == 'trial_config':
+                inverse_config['trial'] = kv['value']
+            elif kv['key'] == 'remote_config':
+                inverse_config['remoteConfig'] = kv['value']
+    elif platform == 'pai':
+        for kv in metadata_config:
+            if kv['key'] == 'pai_config':
+                inverse_config['paiConfig'] = kv['value']
+            elif kv['key'] == 'trial_config':
+                inverse_config['trial'] = kv['value']
+    elif platform == 'kubeflow':
+        for kv in metadata_config:
+            if kv['key'] == 'kubeflow_config':
+                inverse_config['kubeflowConfig'] = kv['value']
+            elif kv['key'] == 'trial_config':
+                inverse_config['trial'] = kv['value']
+    elif platform == 'frameworkcontroller':
+        for kv in metadata_config:
+            if kv['key'] == 'frameworkcontroller_config':
+                inverse_config['frameworkcontrollerConfig'] = kv['value']
+            elif kv['key'] == 'trial_config':
+                inverse_config['trial'] = kv['value']
+    elif platform == 'aml':
+        for kv in metadata_config:
+            if kv['key'] == 'aml_config':
+                inverse_config['amlConfig'] = kv['value']
+            elif kv['key'] == 'trial_config':
+                inverse_config['trial'] = kv['value']
+    else:
+        raise RuntimeError('training service platform not found')
+    return inverse_config
+
 class Config:
     '''a util class to load and save config'''
-    def __init__(self, file_path, home_dir=NNICTL_HOME_DIR):
-        config_path = os.path.join(home_dir, str(file_path))
-        os.makedirs(config_path, exist_ok=True)
-        self.config_file = os.path.join(config_path, '.config')
-        self.config = self.read_file()
+    def __init__(self, experiment_id: str, log_dir: str):
+        self.experiment_id = experiment_id
+        self.conn = sqlite3.connect(os.path.join(log_dir, experiment_id, 'db', 'nni.sqlite'))
+        self.cursor = self.conn.cursor()
+        self.refresh_config()
 
-    def get_all_config(self):
-        '''get all of config values'''
-        return json.dumps(self.config, indent=4, sort_keys=True, separators=(',', ':'))
+    def refresh_config(self):
+        '''refresh to get latest config'''
+        sql = 'select * from ExperimentProfile where id=? order by revision DESC'
+        args = (self.experiment_id,)
+        self.config = config_v0_to_v1(json.loads(self.cursor.execute(sql, args).fetchall()[0][0]))
```

Comment:
Don't use `self.cursor`. One cursor per query.

Reply:
And `.fetchall()[0]` should be `.fetchone()`.

Reply:
fix